CMakeLists.txt

```diff
@@ -120,6 +120,11 @@ xoption(CONFIG_ASAN "Enable AddressSanitizer (ASan)" OFF)
 xoption(CONFIG_MSAN "Enable MemorySanitizer (MSan)" OFF)
 xoption(CONFIG_TSAN "Enable ThreadSanitizer (TSan)" OFF)
 xoption(CONFIG_UBSAN "Enable UndefinedBehaviorSanitizer (UBSan)" OFF)
+if(CMAKE_BUILD_TYPE MATCHES "Debug")
+    xoption(ENABLE_DUMPS "Enable Dumps" ON)
```

Comment:
I'd rather not add a new definition NDEBUG is already a thing.

Reply:
The intent is to allow dumps in release builds or create debug builds without the dump code. IMO this can be quite valuable when debugging issues relating to the dump code. The defaults handle the typical case.

Reply:
I think that's a separate change from what this PR started as. We can take one step at a time.

Reply:
Okay I reverted this change